### DISCUSSION TITLE
Support alternative table formats in markdown

### DISCUSSION THREAD

**Discussion by chipsenkbeil** [squidfunk/mkdocs-material#5677] [Ideas]
From what I can see, Mkdocs Material uses the standard pipe table format.

It'd be great to see other formats as configurable alternatives, or just support parsing multiple types of tables ([like those discussed here](https://stackoverflow.com/questions/57165371/wrap-long-lines-in-markdown-tables)).

The reasoning behind this is the limitations of pipe tables, namely the lack of multiline support, which means that some tables can get **really** long if a column needs a lot of text.

### Pipe table

```
| Name              | Description                                                               | Example           |
| ----------------- | ------------------------------------------------------------------------- | ----------------- |
| Server Version    | The version of the server                                                 | 0.20.0            |
| Protocol Version  | The version of the protocol (different servers can have same protocol)    | 0.1.0             |
| Capabilities      | What features the server has implemented                                  | "search"          |

```

### Grib table

```
+-------------------+--------------------------------------------+-------------------+
| Name              | Description                                | Example           |
+===================+============================================+===================+
| Server Version    | The version of the server                  | 0.20.0            |
+-------------------+--------------------------------------------+-------------------+
| Protocol Version  | The version of the protocol                | 0.1.0             |
|                   | (different servers can have same protocol) |                   |
+-------------------+--------------------------------------------+-------------------+
| Capabilities      | What features the server has implemented   | "search"          |
+-------------------+--------------------------------------------+-------------------+

```

### Multiline table

```
----------------------------------------------------------------------
Name              Description                                 Example
----------------- ------------------------------------------- --------
Server Version    The version of the server                   0.20.0

Protocol Version  The version of the protocol                 0.1.0
                  (different servers can have same protocol)

Capabilities      What features the server has implemented    "search"
----------------------------------------------------------------------
```

**Comment by squidfunk** [comment]
Thanks for suggesting. Indeed, table formatting is not ideal. However, we consider anything that is related to Markdown processing as upstream. Material for MkDocs goes to great lengths to transform what we get from the Markdown parser and MkDocs into a website. Maybe this could be raised to Python Markdown, or there already is an alternative Markdown extension that implements better support for tables? If you make progress on improving table support, feel free to share it here!

**Reply by chipsenkbeil** [reply]
Didn't even think of the markdown extensions approach. Looks like there was some discussion last year about it: https://stackoverflow.com/questions/71039843/support-extended-markdown-table-syntax-with-mkdocs

https://pypi.org/project/markdown-grid-tables/ might be something I could try. Going to take a look.

**Reply by jodygarnett** [reply]
I had a look, since I am running into a large projects with hundreds of grid tables; and I would like to avoid simulating them with definition lists.

While the code exists I cannot quite see how to hook it up.

Installation works:
```
pip3 install markdown-grid-tables
```

But I cannot determine where to connect this in the mkdocs.yml file.

**Reply by kamilkrzyskow** [reply]
Reading the package name in the TOML file I believe that you enable it with:
```yaml
markdown_extensions:
  - markdown_grid_tables
```
I don't see any more config options in the module file.
